.github/workflows/tests.yml

```diff
@@ -487,43 +487,54 @@ jobs:
   #     run: |
   #       pytest ${{ env.PYTEST_PARAMS }} -m "not integration" ${{ env.SUITES_EXCLUDED_FROM_WINDOWS }} test/document_stores/ --document_store_type=pinecone
 
+  # This job collects which Haystack components changed
+  components-changed:
+    runs-on: ubuntu-latest
+    # Set job outputs to the results of the filter step
+    outputs:
+      rest-and-ui: ${{ steps.filter.outputs.rest-and-ui }}
+    steps:
+    # For pull requests it's not necessary to checkout the code
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          rest-and-ui:
+            - 'rest_api/**'
+            - 'ui/**'
 
   rest-and-ui-tests-linux:
     needs:
-      - unit-tests-linux
-      - elasticsearch-tests-linux
+      - mypy
+      - pylint
+      - components-changed
 
     runs-on: ubuntu-latest
-    if: contains(github.event.pull_request.labels.*.name, 'topic:rest_api') || !github.event.pull_request.draft
+    if: ${{ needs.components-changed.outputs.rest-and-ui == 'true' }}
```

Comment:
Sounds good to me! Let's move them in the unit tests block :+1:

Regarding consistency with the docstore tests, shall we do it in a separate PR, or do you want to modify them here? It would be confusing to have some tests activated by labels only and some others by paths.